starknet_py/utils/tuple_dataclass.py

```diff
@@ -0,0 +1,41 @@
+from __future__ import annotations
+from dataclasses import dataclass, fields, make_dataclass
+from typing import Dict, Optional, Tuple
+
+
+@dataclass(frozen=True)
+class TupleDataclass:
+    """
+    Dataclass that behaves like a tuple at the same time.
+    """
+
+    # getattr is called when attribute is not found in object
+    # This way pyright will know that there might be some arguments it doesn't know about and will stop complaining
+    # about some fields that don't exist statically.
+    def __getattr__(self, item):
+        # This should always fail - only attributes that don't exist end up in here.
+        # We use __getattribute__ to get the native error.
+        return super().__getattribute__(item)
+
+    def __getitem__(self, item):
+        field = fields(self)[item]
```

Comment:
How does it work? fields is supposed to return a tuple, shouldn't `item` be a index?

Reply:
Yes, item is an index. This way you can access seconds value with `result[1]`.

Reply:
Can we add type hint to that? 🙏

Reply:
Ok I thought pyright would complain